src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -100,7 +101,15 @@ public RestoreCommand()
 
                 Console.PrintPackageSources(packageSources);
 
-                var localCache = new NuGetv3LocalRepository(packagesDir);
+                var sharedCache = new RestoreCommandSharedCache();
+                sharedCache.LocalCache = new NuGetv3LocalRepository(packagesDir);
```

Comment:
If it doesn't exist a new one is created in RestoreCommand and it will not be shared. It could create it for you in the constructor also. I'm not sure which is clearer, I was trying to avoid a scenario where it creates one that is replaced right away by the shared cache.